test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/ControllerArgumentBinderTests.cs

```diff
@@ -573,6 +574,122 @@ public class ControllerArgumentBinderTests
+                    { fullBindingInfo, differentBindingMetadata, fullBindingInfo },
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(BindingInfoData))]
+        public async Task BindModelAsync_PassesExpectedBindingInfoAndMetadata(
+            BindingInfo parameterBindingInfo,
+            BindingMetadata bindingMetadata,
+            BindingInfo expectedInfo)
+        {
+            // Arrange
+            var metadataProvider = new TestModelMetadataProvider();
+            metadataProvider.ForType<Person>().BindingDetails(binding =>
+            {
+                binding.BinderModelName = bindingMetadata.BinderModelName;
+                binding.BinderType = bindingMetadata.BinderType;
+                binding.BindingSource = bindingMetadata.BindingSource;
+                if (bindingMetadata.PropertyFilterProvider != null)
+                {
+                    binding.PropertyFilterProvider = bindingMetadata.PropertyFilterProvider;
+                }
+            });
+
+            var metadata = metadataProvider.GetMetadataForType(typeof(Person));
+            var binder = new Mock<IModelBinder>();
+            binder
+                .Setup(b => b.BindModelAsync(It.IsAny<ModelBindingContext>()))
+                .Callback((ModelBindingContext context) =>
+                {
+                })
+                .Returns(TaskCache.CompletedTask);
+
+            var factory = new Mock<IModelBinderFactory>(MockBehavior.Strict);
+            factory
+                .Setup(f => f.CreateBinder(It.IsAny<ModelBinderFactoryContext>()))
+                .Callback((ModelBinderFactoryContext context) =>
+                {
+                    Assert.Same(metadata, context.Metadata);
+
+                    Assert.NotNull(context.BindingInfo);
+                    Assert.Equal(expectedInfo.BinderModelName, context.BindingInfo.BinderModelName, StringComparer.Ordinal);
+                    Assert.Equal(expectedInfo.BinderType, context.BindingInfo.BinderType);
+                    Assert.Equal(expectedInfo.BindingSource, context.BindingInfo.BindingSource);
+                    Assert.Same(expectedInfo.PropertyFilterProvider, context.BindingInfo.PropertyFilterProvider);
+                })
+                .Returns(binder.Object);
+
+            var argumentBinder = GetArgumentBinder(factory.Object);
+            var parameterDescriptor = new ParameterDescriptor
+            {
+                BindingInfo = parameterBindingInfo,
+                Name = "",
+                ParameterType = typeof(Person),
+            };
+
+            var controllerContext = GetControllerContext();
+            controllerContext.ActionDescriptor.Parameters.Add(parameterDescriptor);
+
+            // Act
+            var result = await argumentBinder.BindModelAsync(parameterDescriptor, controllerContext);
+
+            // Assert
+            Assert.Equal(ModelBindingResult.Failed(), result);
```

Comment:
Not the purpose of the test. I'm sure you test this elsewhere. Similar [comment](https://github.com/aspnet/Mvc/pull/4724/commits/df0143aba6cd2db0e8ecbe091f270ab92a21d67c#r64973172).